split_to_CMYK.jsx

```diff
@@ -1 +1,124 @@
-﻿var doc = app.activeDocument;

/*var cyanChannel = doc.activeChannels = [doc.channels.getByName('Cyan')];
var magChannel = doc.activeChannels = [doc.channels.getByName('Magenta')];
var yelChannel = doc.activeChannels = [doc.channels.getByName('Yellow')];
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];
*/

/*var CMYK = doc.changeMode(ChangeMode.CMYK);
var RGB = doc.changeMode(ChangeMode.RGB);

var addLayer = doc.artLayers.add();

var renameLayer = doc.artLayers.getByName("Layer 1")
renameLayer.name = "Cyan"

alert(renameLayer)*/



function makeLayers() {
    try {
        if (app.documents.length) {
            var layerColor = ["Cyan","Magenta","Yellow","Black"]
            var doc = activeDocument;
            var layers = doc.artLayers;
            var cmyk = [layerA= layers.add(),
                               layerB= layers.add(),
                               layerC= layers.add(),
                               layerD= layers.add()];

            if (layerColor) {
                cmyk[0].name = layerColor [3];
                cmyk[0].blendMode = BlendMode.MULTIPLY
                cmyk[1].name = layerColor [2];
                cmyk[1].blendMode = BlendMode.MULTIPLY
                cmyk[2].name = layerColor [1];
                cmyk[2].blendMode = BlendMode.MULTIPLY
                cmyk[3].name = layerColor [0];
                cmyk[3].blendMode = BlendMode.MULTIPLY
              
            }
            
            doc.activeLayer = newLayer;
        }
      } catch (e) { 
    }
}
makeLayers ()

/*function splitCyan () {
	var cyan = doc.activeChannels = [doc.channels.getByName('Cyan')];
	app.executeMenuCommand('selectall');
    
}
splitCyan()*/

/*var cyanChannel = doc.activeChannels = [doc.channels.getByName('Cyan')];
var magChannel = doc.activeChannels = [doc.channels.getByName('Magenta')];
var yelChannel = doc.activeChannels = [doc.channels.getByName('Yellow')];
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];*/
\ No newline at end of file
+#target photoshop
+#targetengine "main"
+
+var doc = app.activeDocument;
+
+//switch color mode to get 4 channels
+doc.changeMode(ChangeMode.CMYK);
```

Comment:
Move this line on down as well -- this is another action you're performing within your main set of execution steps so it should come after the function declarations (makes it easier to read the sequence in which your code is performing actions)